.github/workflows/nightly-mjx-test-unit.yaml

```diff
@@ -0,0 +1,137 @@
+    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'failure') && github.event_name != 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo 'Upstream workflow failed, aborting run' && exit 1
+
+  metadata:
+    runs-on: ubuntu-22.04
+    outputs:
+      MJX_IMAGE: ${{ steps.image.outputs.MJX_IMAGE }}
+      PUBLISH: ${{ steps.if-publish.outputs.PUBLISH }}
+      ARTIFACT_NAME: ${{ steps.artifact-name.outputs.ARTIFACT_NAME }}
+    steps:
+      - name: Determine mjx image to use
+        id: image
+        shell: bash -x -e {0}
+        run: |
+          if [[ -z "${{ inputs.MJX_IMAGE }}" ]]; then
+            MJX_IMAGE=${{ env.DEFAULT_MJX_IMAGE }}
+          else
+            MJX_IMAGE=${{ inputs.MJX_IMAGE }}
+          fi
+          echo "MJX_IMAGE=${MJX_IMAGE}" >> $GITHUB_OUTPUT
+
+      - name: Determine artifact name
+        id: artifact-name
+        shell: bash -x -e {0}
+        run: |
+          if [[ -z "${{ inputs.ARTIFACT_NAM }}" ]]; then
+            ARTIFACT_NAME=${{ env.DEFAULT_ARTIFACT_NAME }}
+          else
+            ARTIFACT_NAME=${{ inputs.ARTIFACT_NAME }}
+          fi
+          echo "ARTIFACT_NAME=${ARTIFACT_NAM}" >> $GITHUB_OUTPUT
+
+      - name: Determine whether results will be 'published'
+        id: if-publish
+        shell: bash -x -e {0}
+        run: |
+          echo "PUBLISH=${{ github.event_name == 'workflow_run' || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH) }}" >> $GITHUB_OUTPUT
+
+  runner:
+    uses: ./.github/workflows/_runner_ondemand_slurm.yaml
+    with:
+      NAME: "A100-${{ github.run_id }}"
+      LABELS: "A100:${{ github.run_id }}"
+      TIME: "01:00:00"
+    secrets: inherit
+
+  mjx-unit-test:
+    needs: metadata
+    strategy:
+      fail-fast: false
+      matrix:
+        GPU_ARCH: [V100, A100]
+    # ensures A100 job lands on dedicated runner for this particular job
+    runs-on: [self-hosted, "${{ matrix.GPU_ARCH == 'A100' && format('{0}:{1}', matrix.GPU_ARCH, github.run_id) || matrix.GPU_ARCH }}"]
+    steps:
+      - name: Print environment variables
+        run: env
+
+      - name: Print GPU information
+        run: nvidia-smi  
+
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull MJX image
+        shell: bash -x -e {0}
+        run: |
+          docker pull ${{ needs.metadata.outputs.MJX_IMAGE }}
+
+      - name: MJX speed test
+        shell: bash -x -e {0}
+        continue-on-error: true
+        run: |
+          docker run --gpus=all --shm-size=1g ${{ needs.metadata.outputs.MJX_IMAGE }} bash -ec "mjx-testspeed --mjcf=humanoid/humanoid.xml --batch_size=8192 --unroll=4" | tee -a test-mjx.log
+
+      - name: Upload artifacts
```

Comment:
Please add a step to filter the output and find out lines that looks like
```
humanoid_a100_newton       4.74 s          4.64 s             1 batch_sps=1.73347M jit_time=41.4573
```
and save these as STEP SUMMARY?